### ISSUE TITLE
Tech Draw → Add Centerline to Faces (component)

### ISSUE THREAD

**Issue by kaktusus** [FreeCAD/FreeCAD-translations#48]
```
OS: Debian GNU/Linux 11 (bullseye) (KDE/plasma)
Word size of FreeCAD: 64-bit
Version: 0.20.25065 (Git) AppImage
Build type: Release
Branch: master
Hash: 4de2ad46b2b9c78ac2721a0dbea7c19306261095
Python version: 3.9.4
Qt version: 5.12.9
Coin version: 4.0.0
OCC version: 7.5.2
Locale: Polish/Poland (pl_PL)
```

No translation for the GUI.

![obraz](https://user-images.githubusercontent.com/39754351/122246757-1e791e80-cec7-11eb-8574-5b5d78af5d7d.png)
1.  No edge in selection
No string in Crowdin.
https://crowdin.com/translate/freecad/27911/en-pl?filter=basic&value=0#q=No%20edge%20in%20selection

![obraz](https://user-images.githubusercontent.com/39754351/122247268-84fe3c80-cec7-11eb-9248-bf01589881b0.png)
2. No Part View in Selection
No string in Crowdin.
https://crowdin.com/translate/freecad/27911/en-pl?filter=basic&value=0#q=No%20Part%20view%20in%20selection

additionally, I noticed the existence of a serious bug.

Close the tools :
Add Centerline between 2 Lines
Add Centerline between 2 Points

with the cancel button, **deletes the drawing view**.

**Comment by luzpaz** [comment]
@kaktusus i know this is a massive effort but bugs should be x-posted  to the forum.

**Comment by luzpaz** [comment]
I cant find the strings within the source code??   https://github.com/FreeCAD/FreeCAD/blob/267cb43a328aefeb674104bf5a3cf7cb7cc2076a/src/Mod/TechDraw/Gui/CommandAnnotate.cpp#L675-L774

**Comment by kaktusus** [comment]
> @kaktusus i know this is a massive effort but bugs should be x-posted to the forum.

Yes, I remember you already explained it to me when I asked.

**Comment by kaktusus** [comment]
OS: Debian GNU/Linux bookworm/sid (KDE/plasma)
Word size of FreeCAD: 64-bit
Version: 0.20.28793 (Git) AppImage
Build type: Release
Branch: (HEAD detached at 5386d4e)
Hash: 5386d4ed864cd03edd026e1d3d128cccedf568df
Python 3.9.12, Qt 5.12.9, Coin 4.0.0, OCC 7.5.3
Locale: Polish/Poland (pl_PL)
Installed mods: 
...

After recent updates the problem still persists ...

![obraz](https://user-images.githubusercontent.com/39754351/167263422-d0d60584-9353-431f-b853-5e82ddb28892.png)

but it's a little smaller

**Comment by kaktusus** [comment]
```
OS: Debian GNU/Linux bookworm/sid (KDE/plasma)
Word size of FreeCAD: 64-bit
Version: 0.20.28945 (Git) AppImage
Build type: Release
Branch: (HEAD detached at 0fb7cd9)
Hash: 0fb7cd9758d850f536d571cc094c9833fee18fff
Python 3.9.13, Qt 5.12.9, Coin 4.0.0, OCC 7.5.3
Locale: Polish/Poland (pl_PL)
Installed mods: 
...
```


After recent updates the problem still persists ...

**Comment by Kuzma30** [comment]
@luzpaz  
There are string that need add translation ability (reported here)
https://github.com/FreeCAD/FreeCAD/blob/30f8015e7111a192db55ab66b9c8a5b02d935ef0/src/Mod/TechDraw/Gui/CommandAnnotate.cpp#L1513

https://github.com/FreeCAD/FreeCAD/blob/30f8015e7111a192db55ab66b9c8a5b02d935ef0/src/Mod/TechDraw/Gui/CommandAnnotate.cpp#L1527

They defined as std::stringstream (without translations)

**Comment by chennes** [comment]
I believe this was closed by [29cc754cfdd51974d382c5f7f1be6a96dbf9be3c](https://github.com/FreeCAD/FreeCAD/commit/29cc754cfdd51974d382c5f7f1be6a96dbf9be3c)

**Comment by kaktusus** [comment]
fixed

Version: 0.21.0.33110 (Git) AppImage
